### PR TITLE
NOISSUE - add variable for skip TLS verification on bootstrap

### PR DESCRIPTION
**Pull request by mteodor**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,14 +90,15 @@ Environment:
 |	MF_AGENT_BOOTSTRAP_ID                  | Mainflux bootstrap id                                         |                                   |
 |	MF_AGENT_BOOTSTRAP_KEY                 | Mainflux boostrap key                                         |                                   |
 |	MF_AGENT_BOOTSTRAP_RETRIES             | Number of retries for bootstrap procedure                     | 5                                 |
+| MF_AGENT_BOOTSTRAP_SKIP_TLS            | Skip TLS verification for bootstrap                           | true                              |
```
Comment (manuio) [inline README.md:93]: Fix this please

File: cmd/main.go [hunk 2]
```diff
@@ -33,6 +34,7 @@ const (
 	defBootstrapID                = ""
 	defBootstrapKey               = ""
 	defBootstrapRetries           = "5"
+	defBootstrapSkipTLS           = "false"
```
Comment (manuio) [inline cmd/main.go:37]: Maybe better use `enable` than `skip`

Comment (mteodor) [reply cmd/main.go:37]: this how it is usually called in http library 
```
// Trust the augmented cert pool in our client
	config := &tls.Config{
		InsecureSkipVerify: skipTLS,
		RootCAs:            rootCAs,
	}
```

File: README.md [hunk 3]
```diff
@@ -90,14 +90,15 @@ Environment:
...
 |	MF_AGENT_CONTROL_CHANNEL               | Channel for sending controls, commands                        |                                   |
 |	MF_AGENT_DATA_CHANNEL                  | Channel for data sending                                      |                                   |
 |	MF_AGENT_ENCRYPTION                    | Encryption                                                    | false                             |
 |	MF_AGENT_NATS_URL                      | Nats url                                                      | nats://localhost:4222             |
 |	MF_AGENT_MQTT_USERNAME                 | MQTT username, Mainflux thing id                              |                                   |
 |	MF_AGENT_MQTT_PASSWORD                 | MQTT password, Mainflux thing key                             |                                   |
-|	MF_AGENT_MQTT_SKIP_TLS                 | Skip TLS verification                                         | true                              |
+|	MF_AGENT_MQTT_SKIP_TLS                 | Skip TLS verification for MQTT                                | true                              |
```
Comment (manuio) [inline README.md:101]: Sorry but I still prefer:
```
| Variable                               | Description                                          | Default                               |
|----------------------------------------|------------------------------------------------------|---------------------------------------|
| MF_AGENT_CONFIG_FILE                   | Location of configuration file                       | config.toml                           |
| MF_AGENT_LOG_LEVEL                     | Log level                                            | info                                  |
| MF_AGENT_EDGEX_URL                     | Edgex base url                                       | http://localhost:48090/api/v1/        |
| MF_AGENT_MQTT_URL                      | MQTT broker url                                      | localhost:1883                        |
| MF_AGENT_HTTP_PORT                     | Agent http port                                      | 9000                                  |
| MF_AGENT_BOOTSTRAP_URL                 | Mainflux bootstrap url                               | http://localhost:8202/things/bootstrap|
| MF_AGENT_BOOTSTRAP_ID                  | Mainflux bootstrap id                                |                                       |
| MF_AGENT_BOOTSTRAP_KEY                 | Mainflux boostrap key                                |                                       |
| MF_AGENT_BOOTSTRAP_RETRIES             | Number of retries for bootstrap procedure            | 5                                     |
| MF_AGENT_BOOTSTRAP_SKIP_TLS            | Skip TLS verification for bootstrap                  | true                                  |
| MF_AGENT_BOOTSTRAP_RETRY_DELAY_SECONDS | Number of seconds between retries                    | 10                                    |
| MF_AGENT_CONTROL_CHANNEL               | Channel for sending controls, commands               |                                       |
| MF_AGENT_DATA_CHANNEL                  | Channel for data sending                             |                                       |
| MF_AGENT_ENCRYPTION                    | Encryption                                           | false                                 |
| MF_AGENT_NATS_URL                      | Nats url                                             | nats://localhost:4222                 |
| MF_AGENT_MQTT_USERNAME                 | MQTT username, Mainflux thing id                     |                                       |
| MF_AGENT_MQTT_PASSWORD                 | MQTT password, Mainflux thing key                    |                                       |
| MF_AGENT_MQTT_SKIP_TLS                 | Skip TLS verification for MQTT                       | true                                  |
| MF_AGENT_MQTT_MTLS                     | Use MTLS for MQTT                                    | false                                 |
| MF_AGENT_MQTT_CA                       | Location for CA certificate for MTLS                 | ca.crt                                |
| MF_AGENT_MQTT_QOS                      | QoS                                                  | 0                                     |
| MF_AGENT_MQTT_RETAIN                   | MQTT retain                                          | false                                 |
| MF_AGENT_MQTT_CLIENT_CERT              | Location of client certificate for MTLS              | thing.cert                            |
| MF_AGENT_MQTT_CLIENT_PK                | Location of client certificate key for MTLS          | thing.key                             |
| MF_AGENT_HEARTBEAT_INTERVAL            | Interval in which heartbeat from service is expected | 30s                                   |
| MF_AGENT_TERMINAL_SESSION_TIMEOUT      | Timeout for terminal session                         | 30s                                   |
```

File: README.md [hunk 4]
```diff
@@ -81,32 +81,32 @@ Example configuration:
...
-|	MF_AGENT_HEARTBEAT_INTERVAL            | Interval in which heartbeat from service is expected          | 30s                               |
-|	MF_AGENT_TERMINAL_SESSION_TIMEOUT      | Timeout for terminal session                                  | 30s                               |
+| MF_AGENT_CONFIG_FILE                   | Location of configuration file                                | config.toml                       |
+| MF_AGENT_LOG_LEVEL                     | Log level                                                     | info                              |
+| MF_AGENT_EDGEX_URL                     | Edgex base url                                                | http://localhost:48090/api/v1/    |
+| MF_AGENT_MQTT_URL                      | MQTT broker url                                               | localhost:1883                    |
+| MF_AGENT_HTTP_PORT                     | Agent http port                                               | 9000                              |
+| MF_AGENT_BOOTSTRAP_URL                 | Mainflux bootstrap url                                        | http://localhost:8202/things/bootstrap|
```
Comment (manuio) [inline README.md:89]: This line is not ok

File: pkg/agent/config.go [hunk 5]
```diff
@@ -44,51 +45,45 @@ type MQTTConf struct {
 	PrivKeyPath string          `json:"priv_key_path" toml:"priv_key_path" mapstructure:"priv_key_path"`
 	CA          []byte          `json:"-" toml:"-"`
 	Cert        tls.Certificate `json:"-" toml:"-"`
+	ClientCert  string          `json:"client_cert" toml:"client_cert"`
+	ClientKey   string          `json:"client_key" toml:"client_key"`
+	CaCert      string          `json:"ca_cert" toml:"ca_cert"`
 }
 
-type Heartbeat struct {
+type HeartbeatConf struct {
```
Comment (drasko) [inline pkg/agent/config.go:53]: Sometimes you call this `Conf` and sometime `Config`. Please pick one and stay consistent.

Comment (drasko) [reply pkg/agent/config.go:53]: I think `Config` would be better, looking at `NewConfig()` name...

**Pull request merged**